src/compare-internal.c

```diff
@@ -0,0 +1,38 @@
+#include <stdio.h>
+#include <vips/vips.h>
+
+
+const int STATS_SUM2_COLUMN = 3;
+
+/*
+  Compare images `actual` and `expected` and return mean square error.
+ */
+int Compare(VipsObject *context, VipsImage *actual, VipsImage *expected, double *out) {
+  if (actual->Bands != expected->Bands)
+    return -1;
+
+  if (actual->Type != expected->Type)
+    return -1;
+
+  VipsImage *difference;
+  VipsImage *absolute;
+  VipsImage *stats;
+  if (vips_subtract(actual, expected, &difference, NULL) ||
+      vips_abs(difference, &absolute, NULL) ||
+      vips_stats(absolute, &stats, NULL))
+    return -1;
+
+  vips_object_local(context, difference);
+  vips_object_local(context, absolute);
+  vips_object_local(context, stats);
```

Comment:
Is it safe to free up `stats` here? I am asking because an oversight on my part caused a huge memory leak on our app: https://github.com/gasi/sharp/commit/f620e0bc2c5523481d6cfcf6f682412439dddcfa